owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/CreateRemoteShareOperation.kt

```diff
@@ -0,0 +1,252 @@
+
+/**
+ * Creates a new share.  This allows sharing with a user or group or as a link.
+ *
+ * @author masensio
+ * @author David A. Velasco
+ * @author David González Verdugo
+ */
+class CreateRemoteShareOperation
+/**
+ * Constructor
+ *
+ * @param remoteFilePath Full path of the file/folder being shared. Mandatory argument
+ * @param shareType      0 = user, 1 = group, 3 = Public link. Mandatory argument
+ * @param shareWith      User/group ID with who the file should be shared.  This is mandatory for shareType
+ * of 0 or 1
+ * @param publicUpload   If false (default) public cannot upload to a public shared folder.
+ * If true public can upload to a shared folder. Only available for public link shares
+ * @param password       Password to protect a public link share. Only available for public link shares
+ * @param permissions    1 - Read only Default for public shares
+ * 2 - Update
+ * 4 - Create
+ * 8 - Delete
+ * 16- Re-share
+ * 31- All above Default for private shares
+ * For user or group shares.
+ * To obtain combinations, add the desired values together.
+ * For instance, for Re-Share, delete, read, update, add 16+8+2+1 = 27.
+ */
+    (
+    private val mRemoteFilePath: String,
+    private val mShareType: ShareType,
+    private val mShareWith: String,
+    /**
+     * Upload permissions for the public link (only folders)
+     */
+    private var mPublicUpload: Boolean?,
+    /**
+     * Password to set for the public link
+     */
+    private var mPassword: String?,
+    /**
+     * Access permissions for the file bound to the share
+     */
+    private var mPermissions: Int
+) : RemoteOperation<ShareParserResult>() {
+    private var mGetShareDetails: Boolean = false
+
+    /**
+     * Name to set for the public link
+     */
+    private var mName = ""
+
+    /**
+     * Expiration date to set for the public link
+     */
+    private var mExpirationDateInMillis: Long = 0
+
+    init {
+        mGetShareDetails = false        // defaults to false for backwards compatibility
+    }
+
+    /**
+     * Set name to create in Share resource. Ignored by servers previous to version 10.0.0
+     *
+     * @param name Name to set to the target share.
+     * Null or empty string result in no value set for the name.
+     */
+    fun setName(name: String?) {
+        this.mName = name ?: ""
+    }
+
+    /**
+     * Set password to create in Share resource.
+     *
+     * @param password Password to set to the target share.
+     * Null or empty string result in no value set for the password.
+     */
+    fun setPassword(password: String) {
+        mPassword = password
+    }
+
+    /**
+     * Set expiration date to create in Share resource.
+     *
+     * @param expirationDateInMillis Expiration date to set to the target share.
+     * Zero or negative value results in no value sent for expiration date.
+     */
+    fun setExpirationDate(expirationDateInMillis: Long) {
+        mExpirationDateInMillis = expirationDateInMillis
```

Comment:
the same as before: apply for all setters and getters

Reply:
Sure